pkg/watch/manager.go

```diff
@@ -162,18 +135,19 @@ func (wm *Manager) updateManager() (bool, error) {
 	return true, nil
 }
 
-// updateManagerLoop looks for changes to the watch roster every 5 seconds. This method has a dual
+// Start looks for changes to the watch roster every 5 seconds. This method has a
 // benefit compared to restarting the manager every time a controller changes the watch
 // of placing an upper bound on how often the manager restarts.
-func (wm *Manager) updateManagerLoop(ctx context.Context) {
+func (wm *Manager) Start(done <-chan struct{}) error {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
 	for {
 		select {
-		case <-ctx.Done():
+		case <-done:
 			log.Info("watch manager shutting down")
 			wm.close()
-			return
-		default:
-			time.Sleep(5 * time.Second)
+			return nil
+		case <-ticker.C:
 			if _, err := wm.updateOrPause(); err != nil {
 				log.Error(err, "error in updateManagerLoop")
```

Comment:
return this err

Reply:
I don't think we want to do that, as it is possible for transient errors (e.g. network issue connecting to the API server) to cause an error, which would then cause the server to crash. This would cause the webhook to become unavailable.

I think it's better to have a graceful degradation model where the webhook continues to serve and customize watches could potentially recover on the next restart loop. Detection of this failure state should likely be detected by Prometheus metrics.

WDYT?

Reply:
I see. we definitely dont want this to impact the webhook. So what if it continues to fail to restart the watch manager?

Reply:
Prometheus alerts. We could export metrics for restart failures. Users should also monitor the status fields of constraints/templates to make sure they are operating properly.

Reply:
we should make sure to test this when we add liveness probes. wdyt?

Reply:
Likely not on a continual basis, as failing liveliness probes have the effect of forcing the server to reboot, which is effectively the same as exit-on-failure.

We can test the initial state on startup indirectly by validating cache warming:

1. List all constraints/templates/cached resources on startup
2. Do not report as healthy until we validate those resources have been handled

Reply:
sounds good. @sozercan ^